### PR TITLE
Remove babel polyfill

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

**Comment by azakus** [comment]
This effectively deprecates 0.10 and 0.12, which should require a major version bump and setting ">= 4.0.0" in the `engines` field of `package.json`.

**Comment by azakus** [comment]
Should switch `.travis.yml` node versions to 4, 5, and 6.

File: test/js-parser.html [hunk 1]
```diff
@@ -250,6 +250,7 @@
       setup(function() {
         return hyd.Analyzer.analyze("static/html-behaviors.html").then(function(analyzed){
           analyzer = analyzed;
+          debugger;
```
Comment (rictic) [inline test/js-parser.html]: debugger

File: src/analyzer.ts [hunk 2]
```diff
@@ -257,11 +257,15 @@ export class Analyzer {
   };
 
   load(href: string):Promise<AnalyzedDocument> {
-    return this.loader.request(href).then((content) => {
+    return <any>this.loader.request(href).then((content) => {
```
Comment (rictic) [inline src/analyzer.ts]: Why is this cast needed?

File: src/analyzer.ts [hunk 3]
```diff
@@ -257,11 +257,15 @@ export class Analyzer {
...
       return new Promise<AnalyzedDocument>((resolve, reject) => {
         setTimeout(() => {
           this._content[href] = content;
-          resolve(this._parseHTML(content, href));
+          try {
+            resolve(this._parseHTML(content, href));
+          } catch (err) {
+            reject(err);
+          }
```
Comment (rictic) [inline src/analyzer.ts]: oh nice catch

File: src/analyzer.ts [hunk 4]
```diff
@@ -352,7 +356,7 @@ export class Analyzer {
           .then(function() {return depHrefs;})
           .catch(function(err) {throw err;});
     this.parsedDocuments[href] = parsed.ast;
-    this.html[href] = {
+    this.html[href] = <any>{
```
Comment (rictic) [inline src/analyzer.ts]: why is this cast needed?

File: src/analyzer.ts [hunk 5]
```diff
@@ -395,7 +399,7 @@ export class Analyzer {
         var fixedErr = <LocError>(new Error(message));
         fixedErr.location = {line: line, column: col};
         fixedErr.ownerDocument = script.__ownerDocument;
-        return Promise.reject<DocumentDescriptor>(fixedErr);
+        return Promise.reject<DocumentDescriptor>(<any>fixedErr);
```
Comment (rictic) [inline src/analyzer.ts]: ditto

**Comment by rictic** [comment]
Needs a rebase

**Pull request closed without merging**